`v5.2.0` (#3385)
mui-org/material-ui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 5.2.0

_Dec 9, 2021_

A big thanks to the 5 contributors who made this release possible. Here are some highlights ✨:

### `@mui/x-data-grid@v5.2.0` / `@mui/x-data-grid-pro@v5.2.0`

- 🚀 Introduce the [column pinning](https://mui.com/components/data-grid/columns/#column-pinning) feature (#2946) @m4theushw

>   <img src="https://user-images.githubusercontent.com/5858539/145423491-bf550b77-7744-46e1-8fdb-8be1a9f5b9f5.gif" width="854" height="453" />

Reviewer: ```suggestion
  <img src="https://user-images.githubusercontent.com/42154031/145425635-b6314fbe-2f1e-4b73-908f-33ee1fda20c7.gif" width="964" height="657">
```

![AOlmhqcGME](https://user-images.githubusercontent.com/42154031/145425635-b6314fbe-2f1e-4b73-908f-33ee1fda20c7.gif)

- 🔥 Add ability to disable export options (#3270) @alexfauquette

  You can disable either export options by setting `disableToolbarButton` to `true`.

  ```tsx
  <GridToolbarExport csvOptions={{ disableToolbarButton: true }} />
  ```

  ```tsx
  <GridToolbarExport printOptions={{ disableToolbarButton: true }} />
  ```

> - 🙈 Add `hideDescendantCount` option to Tree Data (#3368) @flaviendelangle

Reviewer 2: Should we be more precise ?

```suggestion
- 🙈 Add a new option to hide the amount of descendant on the grouping cells of the Tree Data (#3368) @flaviendelangle
  
  ```tsx
  <DataGridPro
    treeData
    rows={rows}
    columns={columns}
    groupingColDef={{ hideDescendantCount }}
  />
  ```
```

> - ⚠️ Deprecate the `getValue` param for the `valueGetter` callback (#3314) @flaviendelangle

Reviewer: ```suggestion
- ⚠️ Deprecate the `getValue` param for the `valueGetter` callback (#3314) @flaviendelangle

  Prefer accessing the `row` param:
```

@flaviendelangle do you have a diff?

Reviewer 2: No
We could do something like

  Instead, you can access directly the `row` in the params

  ```diff
  -valueGetter: (params) => `${params.getValue(params.id, 'firstName') || ''} ${params.getValue(params.id, 'lastName') || ''}`
  +valueGetter: (params) => `${params.row.firstName || ''} ${params.row.lastName || ''}`
  ```

Author: Updated

- 📚 Documentation improvements
- 🐞 Bugfixes

#### Changes

- [DataGridPro] Add column pinning (#2946) @m4theushw
- [DataGridPro] Add `hideDescendantCount` option to Tree Data (#3368) @flaviendelangle

> - [DataGridPro] Do not expand row children with Shift + Space (#3380) @flaviendelangle

Reviewer: ```suggestion
- [DataGridPro] Do not expand row children with <kbd>Shift</kbd> + Space (#3380) @flaviendelangle
```

- [DataGridPro] Pass a list of `fields` to the callback version of `groupingColDef` (#3316) @flaviendelangle
- [DataGrid] Deprecate the `getValue` param for the `valueGetter` callback (#3314) @flaviendelangle
- [DataGrid] Add ability to disable export options (#3270) @alexfauquette
- [DataGrid] Filter value are conserved when possible (#3198) @alexfauquette
- [DataGrid] Fix `DatePicker` bug by limiting years to 4 digits (#3222) @alexfauquette
- [DataGrid] Fix column menu position when closing (#3289) @m4theushw

> - [DataGrid] Should not crash when a sort item uses a non-existing column (#3224) @flaviendelangle

Reviewer: ```suggestion
- [DataGrid] Fix to not crash when a sort item uses a non-existing column (#3224) @flaviendelangle
```

### Core

- [core] Add funding field (#3331) @oliviertassinari
- [core] Fix missing LICENSE file (#3330) @oliviertassinari
- [core] Fix release month in CHANGELOG (#3367) @m4theushw
- [core] Fix `yarn prettier` script (#3292) @oliviertassinari
- [core] Improve tests for Tree Data (#3366) @flaviendelangle
- [core] Never import directly from the `__modules__` folder in the `x-data-grid-generator` package (#3379) @flaviendelangle
- [core] Transition to a new StackOverflow tag (#3308) @oliviertassinari

> - [core] Type the api param in callback interfaces (#3315) @flaviendelangle

Reviewer: ```suggestion
- [core] Type the `api` param in callback interfaces (#3315) @flaviendelangle
```

- [core] Update monorepo (#3370) @flaviendelangle
- [core] Use pre-processors for sorting and filtering (#3318) @flaviendelangle

> - [test] ReplaceuseFakeTimers (#3323) @m4theushw

Reviewer: ```suggestion
- [test] Replace `useFakeTimers` (#3323) @m4theushw
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 5.2.0

_Dec 9, 2021_

A big thanks to the 5 contributors who made this release possible. Here are some highlights ✨:

### `@mui/x-data-grid@v5.2.0` / `@mui/x-data-grid-pro@v5.2.0`

- 🚀 Introduce the [column pinning](https://mui.com/components/data-grid/columns/#column-pinning) feature (#2946) @m4theushw

>   <img src="https://user-images.githubusercontent.com/42154031/145425635-b6314fbe-2f1e-4b73-908f-33ee1fda20c7.gif" width="964" height="657">

Reviewer 3: There are two problems with this image. First, by setting the width and height, we prevent the image to always respecting its ratio when the container is resized. It's broken on mobile:

<img width="372" alt="Screenshot 2021-12-09 at 20 57 58" src="https://user-images.githubusercontent.com/3165635/145466922-fc46cb77-938b-45d8-a7cc-6d178b72854a.png">

```suggestion
  <img src="https://user-images.githubusercontent.com/42154031/145425635-b6314fbe-2f1e-4b73-908f-33ee1fda20c7.gif" width="964">
```

---

Second, the image proportion doesn't seem at scale 1:1. I assume that recording was taken on https://mui.com/components/data-grid/demo/. The first column is 200px in the screen recording, while in reality, it's 160px wide. So, in the end, I believe the "right" configuration is:

```suggestion
  <img src="https://user-images.githubusercontent.com/42154031/145425635-b6314fbe-2f1e-4b73-908f-33ee1fda20c7.gif" width="771">
```

I have updated https://github.com/mui-org/material-ui-x/releases/tag/v5.2.0.

I think that developers based their judgment based on what they can see and extrapolate to the rest, so we have to give the changelog some love 😁

Reviewer 3: Side note, flag emoji are not working on Windows. We fixed this problem in the Autocomplete demo of the core, we could consider the same patch here.

<img width="319" alt="Screenshot 2021-12-09 at 21 09 44" src="https://user-images.githubusercontent.com/3165635/145468402-797bccbb-40d1-4a94-85b1-f66aebbec3e4.png">

Reviewer 3: Otherwise, great to see this recording :D
